src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/AddConfigureAwaitAnalyzer.cs

```diff
@@ -0,0 +1,131 @@
+            context.RegisterSyntaxNodeAction(AnalyzeUsingExpression, SyntaxKind.UsingStatement);
+            context.RegisterSyntaxNodeAction(AnalyzeForEachExpression, SyntaxKind.ForEachStatement);
+            context.RegisterSyntaxNodeAction(AnalyzeConfigureAwaitTrue, SyntaxKind.InvocationExpression);
+        }
+
+        private void AnalyzeAwaitExpression(SyntaxNodeAnalysisContext context)
+        {
+            var invocation = (AwaitExpressionSyntax)context.Node;
+            var operation = context.SemanticModel.GetOperation(invocation, context.CancellationToken);
+            if (!(operation is IAwaitOperation awaitOperation)) {
+                return;
+            }
+
+            if (_asyncUtilities.IsTaskType(awaitOperation.Operation.Type)) 
+            {
+                ReportConfigureAwaitDiagnostic(context, awaitOperation);
+            }
+        }
+
+        private void AnalyzeUsingExpression(SyntaxNodeAnalysisContext context) 
+        {
+            if (!(context.SemanticModel.GetOperation(context.Node, context.CancellationToken) is IUsingOperation usingOperation))
+            {
+                return;
+            }
+
+            if (!usingOperation.IsAsynchronous)
+            {
+                return;
+            }
+
+            var resources = usingOperation.Resources;
+            if (_asyncUtilities.IsAsyncDisposableType(resources.Type))
+            {
+                ReportConfigureAwaitDiagnostic(context, resources);
+            }
+        }
+
+        private void AnalyzeForEachExpression(SyntaxNodeAnalysisContext context) 
+        {
+            if (!(context.SemanticModel.GetOperation(context.Node, context.CancellationToken) is IForEachLoopOperation forEachOperation))
+            {
+                return;
+            }
+
+            if (!forEachOperation.IsAsynchronous)
+            {
+                return;
+            }
+
+            var collectionType = forEachOperation.Collection.Type;
+            if (_asyncUtilities.IsAsyncEnumerableType(collectionType)) 
+            {
+                ReportConfigureAwaitDiagnostic(context, forEachOperation.Collection);
+            }
+        }
+
+        private void AnalyzeConfigureAwaitTrue(SyntaxNodeAnalysisContext context)
+        {
+            if (!_asyncUtilities.IsConfigureAwait(context.Node))
+            {
+                return;
+            }
+
+            if (!(context.SemanticModel.GetOperation(context.Node, context.CancellationToken) is IInvocationOperation operation)) 
+            {
+                return;
+            }
+
+            if (!_asyncUtilities.IsConfigureAwait(operation.TargetMethod)) 
+            {
+                return;
+            }
+
+            var constantValue = operation.Arguments.Last().Value?.ConstantValue;
+            if (constantValue != null && constantValue.Value.Value is bool value && value)
+            {
```

Comment:
Can you add a comment as to what this is doing? :)

Reply:
Done